src/index.js

```diff
@@ -16,6 +16,83 @@ document.addEventListener("DOMContentLoaded", function () {
     });
 });
 
+/* The JavaScript Code for Navigation Dynamic Behavior */
+// Our Cache - Stores the partial .HTML pages.
+var partialsCache = {} // Apparently important that this is at the top, or it won't work. 
+/* If no Fragment Identifier is provided then we default to home */
+if (!location.hash) { // Uses the falsy concept
+    location.hash = "#home";
+}
+/* Navigate once to the initial hash value */
+navigate();
+/* Listen for fragment identifier value changes (The # at the end of the URL is the fragment) */
+/* Navigate whenever the fragment identifier value changes */
+window.addEventListener("hashchange", navigate);
+/* Encapsulates an HTTP GET request using XMLHttpRequest
+Fetches the file at the given path, then calls the 
+callback with the content of the file. */
+function fetchFile(path, callback) {
+    // Create a new AJAX request for fetching the partial HTML file.
+    var request = new XMLHttpRequest();
```

Comment:
Isn't this the old way of fetching? Shouldn't we be using the fetch()-function instead, as we do in the rest of the js?

Reply:
I put in a TODO comment about this :)